eventEmitter.js

```diff
@@ -0,0 +1,42 @@
+class EventEmitter {
+   listeners = {};
+     
+   on(eventName, fn) {
+      if(!this.listeners[eventName]) {
+         this.listeners[eventName] = [];
+      } 
+      this.listeners[eventName].push(fn) 
+   }
+        
+   off(eventName, fn) {
+      if (this.listeners[eventName]) {
+         this.listeners[eventName] = this.listeners[eventName].filter((handler) => handler !== fn);
+      }
+   }
+  
+   once(eventName, fn) {
+      const onceWrapper = (...args) => {
+         fn(...args);
+         this.off(eventName, onceWrapper);
+      };
+      this.on(eventName, onceWrapper);
+   }
+  
+   emit(eventName, ...args) {
+      if(this.listeners[eventName]) {
+         this.listeners[eventName].map(fun => args ? fun(...args) : fun())
```

Comment:
I think there is no need to check if args are here, `fun(...args)` should work just fine. But it's better if you test it.